### PR TITLE
cpu: rv64: switch the RV64 backend to runtime ISA dispatch

### PR DESCRIPTION
**Pull request by zhangfeiv0**
## Summary

Now that every RV64 primitive is pure JIT, this PR removes the last remnants of
the intrinsic-era *build-time* ISA gating and turns the backend into genuine
**single-binary runtime dispatch**, mirroring the x64 model (baseline ISA at
compile time + JIT emitting the vector ISA at runtime after `mayiuse()`).

A single `-march=rv64gc` build now runs correctly on both V-capable and non-V
RV64GC hardware: the JIT emits V / Zvfh / Zvfbfwma code only when the running CPU
reports them, and each primitive declines (→ next implementation) when it does
not.

## Motivation

The backend no longer contains a single RVV intrinsic (no `riscv_vector.h`
include anywhere under `src/cpu/rv64`), yet it still carried the build-time
machinery from the intrinsic era:

- `cmake/platform.cmake` probed whether RVV/Zvfh **intrinsics** compile and set
  `DNNL_RISCV_USE_RVV_INTRINSICS` / `DNNL_RISCV_USE_ZVFH_INTRINSICS`.
- `src/cpu/rv64/CMakeLists.txt` excluded every `rvv_*` source when intrinsics
  were unavailable.
- `CPU_INSTANCE_RV64GCV[_ZVFH]` (and the `DNNL_RV64GCV_ONLY[_ZVFH]` macros)
  gated *registration* on those build-time flags, so a `rv64gc` build silently
  dropped all RVV primitives, and a `rv64gcv` build compiled V instructions into
  the reference/driver C++ (so the binary required V to even load).

These are now obsolete and actively misleading. The `TODO`s in
`src/cpu/platform.hpp` and `src/cpu/cpu_engine.hpp` already called for their
removal once all RV64 primitives became JIT.

### Build system — one portable baseline
- `cmake/platform.cmake`: deleted the `riscv_vector.h` compile probe and the
  `DNNL_RISCV_USE_{RVV,ZVFH}_INTRINSICS` definitions; `RV64_MARCH_FLAG` is now
  always `-march=rv64gc`.
- `src/cpu/rv64/CMakeLists.txt`: removed the `rvv_*` source exclusion — all JIT
  sources always compile on the baseline.

### Dead macros removed
- `DNNL_RV64GCV_ONLY` / `DNNL_RV64GCV_ZVFH_ONLY` (`src/cpu/platform.hpp`).
- `CPU_INSTANCE_RV64GCV` / `CPU_INSTANCE_RV64GCV_ZVFH` (`src/cpu/cpu_engine.hpp`).
  The `_ZVFH` variants had zero uses.

### Registration → runtime dispatch
- All 12 `CPU_INSTANCE_RV64GCV(...)` registrations switched to
  `CPU_INSTANCE_RV64(...)` (softmax, convolution ×4, layer/batch norm,
  matmul ×2, inner product ×3).
- The 7 `#if defined(DNNL_RISCV_USE_RVV_INTRINSICS)` include guards around the
  RV64 headers in the implementation lists are now always-on under `#elif
  DNNL_RV64`.
- `src/cpu/gemm/gemm.cpp`: `extended_sgemm`'s RV64 path now routes via a runtime
  `if (mayiuse(v)) return rvv_gemm_f32(...);` and falls through to the portable
  `ref_gemm` otherwise (previously a build-time `#if`, which would have dropped
  the JIT GEMM entirely on a baseline build).

### Correctness — missing runtime ISA gates
Three PDs reach V-emitting JIT but had no `mayiuse(v)` check; they were only safe
because `RV64GCV` hid them on non-V builds. With always-on registration they
would emit illegal instructions on a non-V CPU, so a gate was added to each:
- `rvv_matmul_t` (`src/cpu/rv64/rvv_matmul.hpp`)
- `riscv_gemm_convolution_fwd_t` (`src/cpu/rv64/rvv_gemm_convolution.hpp`)
- `rvv_batch_normalization_fwd_t` (`src/cpu/rv64/rvv_batch_normalization.hpp`)

### Verbose implementation names reflect the runtime ISA
The intrinsic-era `"RISCV64GCV"` / `"RISCV64GCV:gemm"` names were renamed, and the
single-class primitives that span f32/f16(/bf16) now derive the name from the
ISA actually selected (via `JIT_IMPL_NAME_HELPER` + a `cpu_isa_t isa_` set from
the data type in `init()`, the same pattern group normalization already uses):

| primitive | f32 | f16 (Zvfh) | bf16 (Zvfbfwma) |
|---|---|---|---|
| matmul (`rvv_matmul`) | `jit:rvv` | — (f32 only) | — |
| matmul (`rvv_brgemm_matmul`) | `brgemm:rvv` | `brgemm:rvv_zvfh` | `brgemm:rvv_zvfbfwma` |
| softmax | `jit:rvv` | `jit:rvv_zvfh` | — |
| layer norm | `jit_lnorm:rvv` | `jit_lnorm:rvv_zvfh` | — |
| inner product (gemm) | `gemm:rvv` | — (f32 only) | — |
| batch norm | `jit:rvv` | — (f32 only) | — |

Primitives that are f32-only (batch norm, plain matmul, gemm/brgemm inner
product, gemm/brgemm convolution) keep their static `…:rvv` name, which is
correct because no other ISA is ever used. Eltwise / pooling already encode the
ISA via their `<v>` / `<zvfh>` template parameter.

## Behavior change for downstream builds

The default RV64 compile target changes from `-march=rv64gcv[_zvfh]` to
`-march=rv64gc`. This is what makes the binary portable. To let the compiler
auto-vectorize the reference/driver C++ for a V-only deployment, pass
`-DDNNL_ARCH_OPT_FLAGS="-march=rv64gcv"` (or `..._zvfh`) as before; runtime JIT
dispatch is unaffected by that flag.

## Validation

Cross-compiled with `riscv64-linux-gnu-g++-14`, `-march=rv64gc`, `ONEDNN_WERROR=ON`,
Release. **Functional only**

- **Build:** clean, zero warnings under `-Werror`; the whole RV64 backend now
  compiles on the baseline.
- **V present (VLEN=128 and 256):** matmul, convolution (`gemm:rvv`,
  `jit_1x1:rvv`), inner product (`gemm:rvv` f32, `jit:rvv` int8), batch norm,
  softmax, layer norm all dispatch to their RV64 JIT impl with `failed:0`.
- **V absent — same `rv64gc` binary** (`QEMU_CPU=rv64,v=false`): every RV64 JIT
  impl cleanly declines and the reference takes over (`gemm:ref`,
  `ncsp_bnorm:any`, `ref:any`, `gemm_s8u8s32:ref`), `failed:0`, **no illegal
  instruction**.
- **Zvfh (QEMU 11.0.1, f16):** eltwise / pooling → `jit:rvv_zvfh`, softmax →
  `jit:rvv_zvfh`, layer norm → `jit_lnorm:rvv_zvfh`; with `zvfh` off the f16 path
  declines cleanly.
- **Zvfbfwma (QEMU 11.0.1, bf16):** brgemm matmul → `brgemm:rvv_zvfbfwma`
  (f16 → `brgemm:rvv_zvfh`), `failed:0`.

### CODE REVIEW

File: src/cpu/rv64/CMakeLists.txt [hunk 1]
```diff
@@ -19,18 +19,9 @@ file(GLOB_RECURSE SOURCES
...
-    # If required, create an empty source file to prevent it.
-    list(LENGTH SOURCES NUM_OF_SOURCES)
-    if(NUM_OF_SOURCES EQUAL 0)
-        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/empty.cpp "")
-        set(SOURCES ${CMAKE_CURRENT_BINARY_DIR}/empty.cpp)
-    endif()
-endif()
+# The RV64 backend is pure JIT (Xbyak_riscv emits vector code at runtime), so
+# every source compiles for the rv64gc baseline regardless of the V extension.
+# There is no longer an intrinsic-only subset to exclude.
```
Comment (dzarukin) [inline src/cpu/rv64/CMakeLists.txt]: I think we can drop this comment.

**Review by dzarukin — APPROVED** [review]

**Review by zhangjian29 — APPROVED** [review]
